### PR TITLE
!str - Moving the InputStream and OutputStream utilities into Source …

### PR DESCRIPTION
**Pull request by viktorklang**
…and Sink

See #18943

### CODE REVIEW

**Comment by drewhk** [comment]
LGTM

**Comment by viktorklang** [comment]
Unb0rked 2 tests.

**Comment by viktorklang** [comment]
@ktoso This is ready for merge ;)

**Comment by ktoso** [comment]
No need to rush each PR @viktorklang ;-) Deep in debugging things... will review and merge in a bit

File: akka-docs-dev/rst/java/migration-guide-1.0-2.x-java.rst [hunk 1]
```diff
@@ -455,4 +455,87 @@ Example
...
+======================================
+
+Both have been replaced by `Source.inputStream(…)` and `Sink.outputStream(…)` due to discoverability issues.
+
+Update procedure
+----------------
+
+Replace `InputStreamSource.create(` with `Source.inputStream(`
+
+Replace `OutputStreamSink.create(` with `Sink.outputStream(`
```
Comment (ktoso) [inline akka-docs-dev/rst/java/migration-guide-1.0-2.x-java.rst]: rst syntax is: 

```
Replace ``OutputStreamSink.create(`` with ``Sink.outputStream(``
```

Yep, markdown would be simpler, but we're not changing at the moment..

File: akka-docs-dev/rst/scala/code/docs/MigrationsScala.scala [hunk 2]
```diff
@@ -218,6 +218,29 @@ class MigrationsScala extends AkkaSpec {
...
+
+        //#input-output-stream-source-sink
+        val inputStreamSrc = Source.inputStream(() => new SomeInputStream())
+
+        val otherInputStreamSrc = Source.inputStream(() => new SomeInputStream())
+
+        val someOutputStreamSink = Sink.outputStream(() => new SomeOutputStream())
+        //#input-output-stream-source-sink
+
+        val timeout = 1.second
```
Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/MigrationsScala.scala]: move timeout into snippet (so it's clear its Duration and not Timeout_

File: akka-stream/src/main/scala/akka/stream/javadsl/Source.scala [hunk 3]
```diff
@@ -260,8 +260,60 @@ object Source {
    *
    * It materializes a [[Future]] containing the number of bytes read from the source file upon completion.
    */
-  def file(f: File, chunkSize: Int): Source[ByteString, Future[java.lang.Long]] =
+  def file(f: File, chunkSize: Int): javadsl.Source[ByteString, Future[java.lang.Long]] =
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/Source.scala]: :+1:

**Comment by ktoso** [comment]
LGTM after nitpicks (rst syntax and moving timeout a bit lower)

**Comment by viktorklang** [comment]
Done

On Wed, Nov 18, 2015 at 1:30 AM, Konrad Malawski <notifications@github.com>
wrote:

> LGTM after nitpicks (rst syntax and moving timeout a bit lower)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/18947#issuecomment-157554467>.
>



-- 
Cheers,
√

**Comment by ktoso** [comment]
Thanks!

**Pull request merged**